python/packages/autogen-agentchat/src/autogen_agentchat/agents/_assistant_agent.py

```diff
@@ -258,24 +244,54 @@ def __init__(
             for handoff in handoffs:
                 if isinstance(handoff, str):
                     handoff = HandoffBase(target=handoff)
+                if handoff.name in self._handoffs:
+                    raise ValueError(f"Handoff name {handoff.name} already exists.")
                 if isinstance(handoff, HandoffBase):
                     self._handoff_tools.append(handoff.handoff_tool)
                     self._handoffs[handoff.name] = handoff
                 else:
                     raise ValueError(f"Unsupported handoff type: {type(handoff)}")
-        # Check if handoff tool names are unique.
-        handoff_tool_names = [tool.name for tool in self._handoff_tools]
-        if len(handoff_tool_names) != len(set(handoff_tool_names)):
-            raise ValueError(f"Handoff names must be unique: {handoff_tool_names}")
+        if tools is not None:
+            for tool in tools:
+                self.add_tool(tool)
+
+    def add_tool(self, tool: List[Tool | Callable[..., Any] | Callable[..., Awaitable[Any]]]) -> None:
+        new_tool = None
+        if self._model_client.capabilities["function_calling"] is False:
+            raise ValueError("The model does not support function calling.")
+        if isinstance(tool, Tool):
+            new_tool = tool
+        elif callable(tool):
+            if hasattr(tool, "__doc__") and tool.__doc__ is not None:
+                description = tool.__doc__
+            else:
+                description = ""
+            new_tool = FunctionTool(tool, description=description)
+        else:
+            raise ValueError(f"Unsupported tool type: {type(tool)}")
+        self._tools.append(new_tool)
+        # Check if tool names are unique.
+        if new_tool.name in self._tools:
+            raise ValueError(f"Tool names must be unique: {new_tool.name}")
         # Check if handoff tool names not in tool names.
-        if any(name in tool_names for name in handoff_tool_names):
+        handoff_tool_names = [handoff.name for handoff in self._handoffs.values()]
+        if new_tool.name in handoff_tool_names:
             raise ValueError(
-                f"Handoff names must be unique from tool names. Handoff names: {handoff_tool_names}; tool names: {tool_names}"
+                f"Handoff names must be unique from tool names. Handoff names: {handoff_tool_names}; "
+                f"tool names: {new_tool.name}"
             )
-        self._model_context: List[LLMMessage] = []
-        self._reflect_on_tool_use = reflect_on_tool_use
-        self._tool_call_summary_format = tool_call_summary_format
-        self._is_running = False
+
+    def remove_all_tools(self) -> None:
+        """Remove all tools."""
+        self._tools = []
+
+    def remove_tool(self, tool_name: List[str]) -> None:
```

Comment:
```suggestion
    def remove_tool(self, tool_name: str) -> None:
```

Reply:
@ekzhu fixed in 14e4dc1e98eaaf00926b0b8ce09430fac6465e30